cmd/command_flags.go

```diff
@@ -4,92 +4,175 @@ import "github.com/urfave/cli/v2"
-func outputFlag() *cli.StringFlag {
+// OutputFlag helps to control the style of output
+func OutputFlag() *cli.StringFlag {
 	return &cli.StringFlag{
-		Name:  "output",
-		Usage: "control output, accepts 'print' or 'monitor'",
-		Value: "print",
+		Name:    "output",
+		Usage:   "control output, accepts 'print' or 'monitor'",
+		Value:   "print",
+		EnvVars: []string{"OUTPUT"},
 	}
 }
 
-func keyName() *cli.StringFlag {
+// KeyName indicates the name of the key
+func KeyName() *cli.StringFlag {
 	return &cli.StringFlag{
 		Name:    "key.name",
 		Aliases: []string{"kn"},
 		Usage:   "name of the key used in the keystore",
+		EnvVars: []string{"KEY_NAME", "KN"},
 	}
 }
 
-func keyType() *cli.StringFlag {
+// KeyType indicates the type of key
+func KeyType() *cli.StringFlag {
 	return &cli.StringFlag{
 		Name:    "key.type",
 		Aliases: []string{"kt"},
-		Usage:   "type of key: ed25519, ecdsa, rsa",
+		Usage:   "type of key: ed25519, ecdsa, rsa, secp256k1",
+		EnvVars: []string{"KEY_TYPE", "KT"},
+	}
+}
+
+// KeySize indicates the size of a key, default size suitable for all but RSA
+func KeySize() *cli.IntFlag {
+	return &cli.IntFlag{
+		Name:    "key.size",
+		Value:   256,
+		Aliases: []string{"ks"},
+		Usage:   "size of key in bytes",
+		EnvVars: []string{"KEY_SIZE", "KS"},
+	}
+}
+
+// MnemonicFlag allows saving data as a mnemonic
+func MnemonicFlag() *cli.StringFlag {
+	return &cli.StringFlag{
+		Name:    "save.mnemonic",
+		Aliases: []string{"sm"},
+		Usage:   "save mnemonic to `PATH` if not empty",
+		Value:   "",
+		EnvVars: []string{"SAVE_MNEMONIC", "SM"},
+	}
+}
+
+// InputFileFlag allows reading data from a file
+func InputFileFlag() *cli.StringFlag {
+	return &cli.StringFlag{
+		Name:    "input.file",
+		Aliases: []string{"in.fi", "if"},
+		Usage:   "load data contained in file at `PATH`",
+		EnvVars: []string{"INPUT_FILE", "IF"},
+	}
+}
+
+// IsHexEncodedFlag indicates if the data is hex encoded
+func IsHexEncodedFlag() *cli.BoolFlag {
+	return &cli.BoolFlag{
+		Name:    "hex.encoded",
+		Value:   false,
+		Usage:   "whether or not the key has been hex encoded",
+		EnvVars: []string{"HEX_ENCODED"},
+	}
+}
+
+// IsMnemonicEncodedFlag indicates if the data is encoded as mnemonic
+func IsMnemonicEncodedFlag() *cli.BoolFlag {
+	return &cli.BoolFlag{
+		Name:    "mnemonic.encoded",
+		Value:   true,
+		Usage:   "whether or not the key has been converted into a mnemonic",
+		EnvVars: []string{"MNEMONIC_ENCODED"},
 	}
 }
 
+// P2pFlags are used to control p2p stream
 // takes in an argument which is a command that should be
 // loaded with a default value. This is appended to the default
 // p2p command flag list
-func p2pFlags(cmdFlag *cli.StringFlag) []cli.Flag {
+func P2pFlags(cmdFlag *cli.StringFlag) []cli.Flag {
 	if cmdFlag.Value == "" {
 		panic("flag value is nil")
 	}
 	return append([]cli.Flag{
 		&cli.BoolFlag{
-			Name: "all",
+			Name:    "all",
+			Usage:   "close all listeners. used by: close",
+			EnvVars: []string{"ALL"},
```

Comment:
I think we should prefix the environmental variables with somthing like, `TEX_` other wise some names are very generic and could collide.

Reply:
good call